### PR TITLE
ISSUE-5 Setting Armeria server

### PR DESCRIPTION
**Pull request by AmosDoan**
**ISSUE**
- https://github.com/tokyoprogrammer/aqitracerAPI/issues/5
- https://github.com/tokyoprogrammer/aqitracerAPI/issues/6

**Summary**
- Setting Armeria server using annotated service
   - https://line.github.io/armeria/server-annotated-service.html
- Setting logback-spring.xml

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,9 +22,30 @@ repositories {
...
+    imports {
+        mavenBom 'com.linecorp.armeria:armeria-bom:0.79.0'
+    }
+}
+
 dependencies {
-    implementation 'org.springframework.boot:spring-boot-starter-web'
+    // Armeria
+    compile 'com.linecorp.armeria:armeria-spring-boot-starter'
+
```
Comment (YoseobKim) [inline build.gradle:34]: Does armeria have full compatibility with normal pivotal spring boot(or pivotal product family)? It seems the version naming and way of version control of the package are little bit different from original spring?

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java [hunk 2]
```diff
@@ -1,13 +1,47 @@
...
+    @Get
+    @Path("/healthCheck")
+    public HttpResponse healthCheck() {
+        return HttpResponse.of(HttpStatus.OK, MediaType.JSON_UTF_8, "OK");
+    }
+
+    @Get
+    @Path("/api/v1/aqi")
+    public String getAqi(@Param("latlng") String latlng, @Param("provider") AQI_PROVIDER provider) {
+        return ImmutableMap.of("coordinate", latlng, "provider", provider).toString();
```
Comment (YoseobKim) [inline src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java:39]: We'd better have more discussion about parameters for this API.

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java [hunk 3]
```diff
@@ -1,13 +1,47 @@
...
+    }
+
+    @Get
+    @Path("/api/v1/etc")
+    public String getEtc(@Param("type") String type) {
```
Comment (YoseobKim) [inline src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java]: Why don't you use ETC_TYPE for this parameter like above? Is there specific reason for this?

**Review by YoseobKim — COMMENTED** [review]
Please check this comments. Thank you :)

**Review by jb-hy — APPROVED** [review]

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java [hunk 4]
```diff
@@ -1,13 +1,47 @@
...
+        EUROPE,
+    };
+
+    private enum ETC_TYPE {
+        WALK,
+        CULTURE
+    };
+
+    @Get
+    @Path("/healthCheck")
```
Comment (jb-hy) [inline src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java:31]: Does Armeria server provide reactive endpoints like Spring WebFlux? I guess the Armeria annotations such as `@Get` and `@Path` are not so different from `@GetMapping`, am I correct? If not, could you explain more about it?

**Review by jb-hy — APPROVED** [review]
I think it is no problem to merge these changes. Thank you so much!

File: build.gradle (hunk 1 above)
Comment (AmosDoan) [reply build.gradle:34]: Yes. Armeria have full compatibility with spring boot. As you concerned, Armeria would be released for every 2 weeks as consider spring boot compatibility, so it is okay.

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java (hunk 4 above)
Comment (AmosDoan) [reply src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java:31]: As you said, Armeria supports the integration with Spring WebFlux. Please refer to it https://line.github.io/armeria/advanced-spring-webflux-integration.html?highlight=webflux I think we can consider reactive stream 👍

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java (hunk 3 above)
Comment (AmosDoan) [reply src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java]: Oh I forgot to define it. thanks for comment!

File: src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java (hunk 2 above)
Comment (AmosDoan) [reply src/main/java/com/tokyoprogrammer/aqitracer/controller/AqitracerController.java:39]: Yes. how about handling about API spec. for next issue? I created new issue https://github.com/tokyoprogrammer/aqitracerAPI/issues/9

**Review by YoseobKim — APPROVED** [review]
LGTM. (could you squash the commits?)

**Pull request merged**